framework/src/org/checkerframework/common/util/count/TypeStatistics.java

```diff
@@ -0,0 +1,79 @@
+package org.checkerframework.common.util.count;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.NewArrayTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import org.checkerframework.framework.source.SourceChecker;
+import org.checkerframework.framework.source.SourceVisitor;
+import org.checkerframework.javacutil.AnnotationProvider;
+
+/**
+ * An annotation processor for listing the facts about types and array accesses locations of
+ * annotations. To invoke it, use
+ *
+ * <pre>
+ * javac -proc:only -processor org.checkerframework.common.util.count.TypeStatistics <em>MyFile.java ...</em>
+ * </pre>
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class TypeStatistics extends SourceChecker {
+
+    int generics = 0;
+    int arrayAccesses = 0;
+
+    @Override
+    public void typeProcessingOver() {
+        System.out.printf("Found %d generic type uses.\n", generics);
+        System.out.printf("Found %d array accesses and creations.\n", arrayAccesses);
+    }
+
+    @Override
+    protected SourceVisitor<?, ?> createSourceVisitor() {
+        return new Visitor(this);
+    }
+
+    class Visitor extends SourceVisitor<Void, Void> {
+
+        public Visitor(TypeStatistics l) {
+            super(l);
+        }
+
+        @Override
+        public Void visitClass(ClassTree tree, Void p) {
+            if (shouldSkipDefs(tree)) {
+                // Not "return super.visitClass(classTree, p);" because that would
+                // recursively call visitors on subtrees; we want to skip the
+                // class entirely.
+                return null;
+            }
+            return super.visitClass(tree, p);
+        }
+
+        @Override
+        public Void visitNewArray(NewArrayTree node, Void aVoid) {
+            arrayAccesses += node.getDimensions().size();
```

Comment:
This doesn't count arrays created using initializers.  Is that what we want?

Reply:
I think that's OK.  We really want to count the number of indexes, and the argument i to `new T[i]` is index-like but there is no index-like thing in `a = {0, 1, 2}`.  We should document what the class counts, though.